staging/src/k8s.io/client-go/restmapper/shortcut.go

```diff
@@ -43,7 +43,11 @@ func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInt
 
 // KindFor fulfills meta.RESTMapper
 func (e shortcutExpander) KindFor(resource schema.GroupVersionResource) (schema.GroupVersionKind, error) {
-	return e.RESTMapper.KindFor(e.expandResourceShortcut(resource))
+	gvk, err := e.RESTMapper.KindFor(e.expandResourceShortcut(resource))
+	if meta.IsNoMatchError(err) {
```

Comment:
would have expected to call `e.expandResourceShortcut(resource)` first and rerun on error or empty result. Isn't `e.expandResourceShortcut` failing?

Reply:
Currently, `expandResourceShortcut` will just return resource as is if it doesn't find shortcut, but yeah I could return error or empty and based on that re-run expansion. lemme fix that...

Reply:
@soltysh hi, thanks for getting this PR started.  There's some interest in getting this going.  We're trying to fix a large number flaky tests.  Can you get the above patch going?